Null check m_currentOpenDBRequest in notifyCurrentRequestConnectionClosedOrFiredVersionChangeEvent

achristensen07 opened the pull request:
#### 2eed0466d7087a67e3b65404c23d3e1622e6ae15
<pre>
Null check m_currentOpenDBRequest in notifyCurrentRequestConnectionClosedOrFiredVersionChangeEvent
<a href="https://bugs.webkit.org/show_bug.cgi?id=263474">https://bugs.webkit.org/show_bug.cgi?id=263474</a>
rdar://116852330

Reviewed by Sihui Liu.

Crash logs indicate this is necessary.

* Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabase.cpp:
(WebCore::IDBServer::UniqueIDBDatabase::didFireVersionChangeEvent):

Canonical link: <a href="https://commits.webkit.org/269623@main">https://commits.webkit.org/269623@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5616489bb8170ac1c2c0812cb2ddd310697d7bbc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23025 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1105 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24138 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24940 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21298 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/23290 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/2340 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23563 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/22165 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23267 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/548 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19975 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25793 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/430 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20858 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27029 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20812 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21122 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/24895 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/535 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18327 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/460 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/943 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2934 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/731 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabase.cpp:
```diff
@@ -546,6 +546,9 @@ void UniqueIDBDatabase::didFireVersionChangeEvent(UniqueIDBDatabaseConnection& c
         }
     }
 
+    if (!m_currentOpenDBRequest)
```
Should this be inside notifyCurrentRequestConnectionClosedOrFiredVersionChangeEvent() ? Because it is weird to null check something that is not used later in the function.

szewai replied:
Yes, it seems we can just move this to `notifyCurrentRequestConnectionClosedOrFiredVersionChangeEvent` and drop the `ASSERT(m_currentOpenDBRequest);` in it; then we could remove the null check in `UniqueIDBDatabase::connectionClosedFromClient` too.

achristensen07 committed:
Null check m_currentOpenDBRequest in notifyCurrentRequestConnectionClosedOrFiredVersionChangeEvent
https://bugs.webkit.org/show_bug.cgi?id=263474
rdar://116852330

Reviewed by Sihui Liu.

Crash logs indicate this is necessary.

* Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabase.cpp:
(WebCore::IDBServer::UniqueIDBDatabase::didFireVersionChangeEvent):

Canonical link: https://commits.webkit.org/269623@main
